lms/djangoapps/grades/signals/handlers.py

```diff
@@ -183,3 +192,31 @@ def recalculate_course_grade(sender, course, course_structure, user, **kwargs):
     Updates a saved course grade.
     """
     CourseGradeFactory().update(user, course, course_structure)
+
+
+def _emit_problem_submitted_event(kwargs):
+    """
+    Emits a problem submitted event only if
+    there is no current event transaction type,
+    i.e. we have not reached this point in the
+    code via a rescore or student state deletion.
+    """
+    root_type = get_event_transaction_type()
+
+    if not root_type:
+        root_id = get_event_transaction_id()
+        if not root_id:
+            create_new_event_transaction_id()
```

Comment:
You can update `create_new_event_transaction_id` to return the newly generated id.  Then, you can update the code below to use the `root_id` variable instead of getting it from the request_cache again.